linode/manager
# change: [M3-7289] – Payload refinements in "Assign Linodes" flow

## Description 📝
Currently, VPC interfaces added via the Assign Linodes flow are not marked as `primary` and are not 1:1 NAT'ed. This results in a sub-optimal and possibly non-functional configuration.

## Changes  🔄
- Add `primary: true` and `nat_1_1: 'any'` to the interface payload
- If the config being modified only had an implicit public interface previously (i.e., the `interfaces` array comes back from the API as empty), explicitly add a public interface in eth0

## How to test 🧪

### Prerequisites
- An account with the proper VPC tags
- Pointed at the `dev` environment

### Verification steps 
With the Network tab of Dev Tools open...
- Create a new Linode with no VPC or VLAN assigned
  - Go to a VPC in the same region as the newly-created linode with a subnet and enter the "Assign Linodes" flow. Select the newly-created linode and assign it. Observe: two requests to `/interfaces`, the first for adding an explicit public interface, and the second for the VPC interface (confirm it has `primary: true` and `nat_1_1: 'any'`)

For other possible permutations (e.g., a linode with a VLAN already assigned), confirm that going through the "Assign Linodes" flow results in a VPC interface being added with `primary: true` and `nat_1_1: 'any'` in the payload. No explicit public interfaces should be in the payload outside of the scenario detailed above.

## As an Author I have considered 🤔

*Check all that apply*
